source/vibe/core/drivers/libevent2_tcp.d

```diff
@@ -83,9 +83,12 @@ package final class Libevent2TCPConnection : TCPConnection {
 
 		void* ptr;
 		if( ctx.remote_addr.family == AF_INET ) ptr = &ctx.remote_addr.sockAddrInet4.sin_addr;
-		else ptr = &ctx.remote_addr.sockAddrInet6.sin6_addr;
-		evutil_inet_ntop(ctx.remote_addr.family, ptr, m_peerAddressBuf.ptr, m_peerAddressBuf.length);
-		m_peerAddress = cast(string)m_peerAddressBuf[0 .. m_peerAddressBuf[].indexOf('\0')];
+		else if (ctx.remote_addr.family != AF_UNIX ) ptr = &ctx.remote_addr.sockAddrInet6.sin6_addr;
+		else ptr = &ctx.remote_addr.sockAddrUnix.sun_path;
+
+		if (evutil_inet_ntop(ctx.remote_addr.family, ptr, m_peerAddressBuf.ptr, m_peerAddressBuf.length) !is null)
+			m_peerAddress = cast(string)m_peerAddressBuf[0 .. m_peerAddressBuf[].indexOf('\0')];
+		// TODO: do we need to set m_peerAddress in else case? (when family is AF_UNIX)
```

Comment:
The peer address is used in some places to construct URLs, but I'm not sure how much sense it makes in this context. Probably safe to skip it for now.